lib/sentry/envelope.ex

```diff
@@ -34,6 +34,41 @@ defmodule Sentry.Envelope do
     }
   end
 
+  @doc """
+  Creates a new envelope containing the client report.
+  since: "10.8.0"
+  """
+  @spec from_client_report(ClientReport.t()) :: t()
+  def from_client_report(%ClientReport{} = client_report) do
+    %__MODULE__{
+      event_id: UUID.uuid4_hex(),
+      items: [client_report]
+    }
+  end
+
+  def get_data_category(type) do
+    if is_binary(type) do
+      type
```

Comment:
When will this ever be a string? We talked about functions having a small "type" surface but this function is still accepting `binary | Attachment.t | ...`. Can we make it so this only accepts a struct OR a binary?

Also, can you add typespecs and tests for this?